File: 61. Rotate List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def rotateRight(self, head, k):
        """
        :type head: ListNode
        :type k: int
        :rtype: ListNode
        """
        if not head:
            return []
        length = 1
        p = head
        while p.next:
            p = p.next
            length += 1
        p.next = head
        k %= length
        for i in xrange(length - k):
            p = p.next
        dummy = p.next
        p.next = None
        return dummy

File: 61. Rotate List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var rotateRight = function(head, k) {
    if (head === null)
        return [];
    let p = head;
    let length = 1;
    while (p.next !== null)
    {
        length += 1;
        p = p.next;
    }
    p.next = head;
    k %= length;
    for (let i = 0; i < length - k; i ++)
        p = p.next;
    dummy = p.next;
    p.next = null;
    return dummy;
};
